File: Flutter/callkit/ios/Classes/CallkitPlugin.m
// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

#import "CallkitPlugin.h"

#if __has_include(<netease_callkit/netease_callkit-Swift.h>)
#import <netease_callkit/netease_callkit-Swift.h>
#else
// Support project import fallback if the generated compatibility header
// is not copied when this plugin is created as a library.
// https://forums.swift.org/t/swift-static-libraries-dont-copy-generated-objective-c-header/19816
#import "netease_callkit-Swift.h"
#endif

@implementation CallkitPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
  [SwiftCallkitPlugin registerWithRegistrar:registrar];
}
@end

File: Flutter/callkit/ios/Classes/CallkitPlugin.swift
// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

import Flutter
import UIKit

public class SwiftCallkitPlugin: NSObject, FlutterPlugin {
  private var linker: FLTLinker?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let instance = SwiftCallkitPlugin()
    instance.linker = FLTLinker()
    instance.linker?.onAttachedToEngine(registrar)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }

  public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
    linker?.onDetachedToEngine(registrar)
    linker = nil
  }
}
